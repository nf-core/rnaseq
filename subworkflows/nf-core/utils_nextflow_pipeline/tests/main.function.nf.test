
nextflow_function {

    name "Test Functions"
    script "subworkflows/nf-core/utils_nextflow_pipeline/main.nf"
    config "subworkflows/nf-core/utils_nextflow_pipeline/tests/nextflow.config"
    tag 'subworkflows'
    tag 'utils_nextflow_pipeline'
    tag 'subworkflows/utils_nextflow_pipeline'

    test("Test Function getWorkflowVersion") {

        function "getWorkflowVersion"

        then {
            assertAll(
                { assert function.success },
                { assert snapshot(function.result).match() }
            )
        }
    }

    test("Test Function dumpParametersToJSON") {

        function "dumpParametersToJSON"

        when {
            function {
                """
                // define inputs of the function here. Example:
                input[0] = "$outputDir"
                """.stripIndent()
            }
        }

        then {
            assertAll(
                { assert function.success }
            )
        }
    }

    test("Test Function checkCondaChannels") {

        function "checkCondaChannels"

        then {
            assertAll(
                { assert function.success },
                { assert snapshot(function.result).match() }
            )
        }
    }
}