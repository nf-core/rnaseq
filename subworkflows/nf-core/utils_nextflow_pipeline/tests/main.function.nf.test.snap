{
    "Test Function getWorkflowVersion": {
        "content": [
            "v9.9.9"
        ],
        "timestamp": "2024-01-19T11:32:36.031083"
    },
    "Test Function checkCondaChannels": {
        "content": null,
        "timestamp": "2024-01-19T11:32:50.456"
    }
}